internal/handlers/common/aggregations/count.go

```diff
@@ -30,7 +30,7 @@ type count struct {
 }
 
 // newCount creates a new $count stage.
-func newCount(stage *types.Document) (Stage, error) {
+func newCount(stage *types.Document, fetcher Fetcher) (Stage, error) {
```

Comment:
Transfering fetcher to all the functions (even the ones where it's not needed) doesn't look very flexible.
Should we extend the `newX` functions with something more flexible? 

For example,

`newX (stage *types.Document, opts ...*stageOptions)`

and then `fetcher` could be one of such options that we pass only to the function that is needed?

Though, I'm not sure if it's possible to do it this way -- I see in the handler we iterate through `for i, d := range stagesDocs {` and then for each stage we call that `new` with fetcher... 

I think I need more time to think how it could be improved :)

Reply:
Good point. for `newX(...)` let me see if I can think of a way to do it. :thinking: 

For `opts ...*stageOptions`, I thought of that approach, of passing values but we have to can't pass `pg dbPool` or `tigris dbPool` so we will have to pass an interface in it anyway for now. So I just have interface without wrapping in another structure.